pkg/apis/kf/v1alpha1/app_defaults.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import "context"
+
+// SetDefaults implements apis.Defaultable
+func (k *App) SetDefaults(ctx context.Context) {
+	k.Spec.SetDefaults(ctx)
+}
+
+// SetDefaults implements apis.Defaultable
+func (k *AppSpec) SetDefaults(ctx context.Context) {
+	// XXX: currently no defaults to set
```

Comment:
I think instance count and status could be set here if they weren't already unless that's for a future PR.

Reply:
I'm saving defaults for later PRs